bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3Storage.java

```diff
@@ -194,24 +194,36 @@ public boolean supportsTruncation() {
      * Iterator for segments in ExtendedS3Storage.
      */
     public class ExtendedS3SegmentIterator implements Iterator<SegmentProperties> {
```

Comment:
1. This class does not need to be public. 
2. Can it be static?
3. To better organize the code, please move the class at the bottom of this file. Do not mix methods and inner classes together.